pandas/core/reshape/concat.py

```diff
@@ -362,17 +364,22 @@ def __init__(
         self.objs = objs
 
         # Standardize axis parameter to int
-        if isinstance(sample, Series):
+        # TODO: Should this really require a class import?
+        """
+        if isinstance(sample, ABCSeries):
```

Comment:
This is maybe a relic of the old panel days but as of now it seems a little overkill to import the generic class definition to resolve "index" and "columns" to their respective axis numbers. For now I just did this in the function body directly below commented code, but this probably belongs somewhere else as a utility function?

Reply:
my intuition is that you're right here.  A lot of _axis_reversed stuff feels leftover